pubsublite-spark-sql-streaming/src/main/java/com/google/cloud/pubsublite/spark/PslContinuousReader.java

```diff
@@ -59,8 +60,9 @@ public PslContinuousReader(
 
   @Override
   public Offset mergeOffsets(PartitionOffset[] offsets) {
-    assert SparkPartitionOffset.class.isAssignableFrom(offsets.getClass().getComponentType())
-        : "PartitionOffset object is not assignable to SparkPartitionOffset.";
+    Preconditions.checkState(
+        SparkPartitionOffset.class.isAssignableFrom(offsets.getClass().getComponentType()),
```

Comment:
here and elsewhere: If the thing you're checking is a precondition on an argument, checkArgument instead.